### PR TITLE
Backups

### PR DESCRIPTION
**Pull request by tpetracca**
@joelea I wanted to break this into pieces but it was going to be time consuming so I figured I'd put this here first for feedback and see if you were just find doing the whole thing at once.

I got a little trigger happy on the whole AbstractTimestampCommand refactor because the amount of duplicated code I had was sad.  I'm happy to rewind some of that if people don't like it.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/646)

### CODE REVIEW

**Comment by tpetracca** [comment]
This addresses #587

File: atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/timestamp/FetchTimestamp.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    @Override
+    protected boolean requireTimestamp() {
+        return false;
+    }
+
+	@Override
+	protected int executeTimestampCommand(AtlasDbServices services) {
+	    String name;
+	    if (immutable) {
+    	    timestamp = services.getTransactionManager().getImmutableTimestamp();
```
Comment (joelea) [inline atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/timestamp/FetchTimestamp.java]: nit: weird formatting.

File: atlasdb-cli/src/test/java/com/palantir/atlasdb/cli/command/TestTimestampCommand.java [hunk 2]
```diff
@@ -56,46 +78,120 @@ public TestAtlasDbServices connect(ServicesConfigModule servicesConfigModule) {
         };
     }
 
-    private SingleBackendCliTestRunner makeRunner(String... args) {
-        return new InMemoryTestRunner(TimestampCommand.class, args);
+    @AfterClass
```
Comment (joelea) [inline atlasdb-cli/src/test/java/com/palantir/atlasdb/cli/command/TestTimestampCommand.java]: I have no idea what any of these tests are doing. It's not something I really expect you to clean up - but I also can't really review them.

File: atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/timestamp/FetchTimestamp.java (hunk 1 above)
Comment (tpetracca) [inline atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/timestamp/FetchTimestamp.java]: eclipse decided to randomly allow tabs again *sigh*
Is this something that can easily be added to checkstyle?  Basically disallow tabs, only allow 4-spaces?

File: atlasdb-cli/src/test/java/com/palantir/atlasdb/cli/command/TestTimestampCommand.java (hunk 2 above)
Comment (tpetracca) [inline atlasdb-cli/src/test/java/com/palantir/atlasdb/cli/command/TestTimestampCommand.java]: Going to try and add some javadoc to explain what's I'm testing here

**Comment by tpetracca** [comment]
Apologies for the force push; I used `git pull --rebase origin develop` instead of merging origin/develop into this branch so when I had to fix a conflict it became impossible (I think?) for me to push the normal way because it thought I was behind origin/backups

**Comment by clockfort** [comment]
Reviewed 12 of 16 files at r1, 7 of 9 files at r2, 1 of 1 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/646)*

**Comment by clockfort** [comment]
I squashed your commits and will merge this.

**Pull request merged**